rasa/core/policies/flow_policy.py

```diff
@@ -695,17 +695,31 @@ def _run_step(
                     and previous_flow_step
                     and current_frame.frame_type == StackFrameType.CORRECTION
                 ):
-                    # TODO: we need to figure out how to actually
-                    #    "undo" the changed slots
                     corrected_slots = tracker.get_slot(CORRECTED_SLOTS_SLOT)
                     if corrected_slots:
-                        self._correct_flow_position(
-                            corrected_slots, previous_flow_step, previous_flow, tracker
-                        )
-                    else:
-                        # TODO: we need to figure out how to actually "undo" the
-                        #    changed slots
-                        pass
+                        if (
+                            tracker.latest_action_name
+                            == "utter_corrected_previous_input"
+                        ):
+                            self._correct_flow_position(
+                                corrected_slots,
+                                previous_flow_step,
+                                previous_flow,
+                                tracker,
+                            )
+                        else:
+                            found_first_event = False
```

Comment:
what's this used for?

Reply:
This is to fetch the second-to-last slot value for a specific slot from the tracker

```
user: I want to transfer 50 USD
bot: ..
user: ahh scratch that, I want to send 100
bot: Do you want to update your information?
user: No, keep the previous information
```

At this point, the `transfer_money_amount_of_money` slot is already set to 100, to revert back to 50, I search for the previous value. Hence, I used a flag variable to fetch the second-to-last value

Reply:
Ah now I get it yes makes sense. but fails if there is more than one slot int `corrected_slots`

Reply:
Is there a possibility of having multiple slot corrections from a single-user message? 

The existing implementation was resetting the `corrected_slots` to `None` in the `pattern_correction` flow

Reply:
Thanks a lot, Tom, I have now changed the code. Also, added a test here: https://github.com/RasaHQ/financial-demo-flows-llms/pull/28/files